Add NuGet 6.13 Release Notes (#3390)
NuGet/docs.microsoft.com-nuget: docs/release-notes/NuGet-6.13.md

---
title: NuGet 6.13 Release Notes
description: Release notes for NuGet 6.13 including new features, bug fixes, and DCRs.

> author: [nyenework](https://github.com/Nigusu-Allehu)

Reviewer: I think you're supposed to just provide that alias.

title: NuGet 6.13 Release Notes
description: Release notes for NuGet 6.13 including new features, bug fixes, and DCRs.
author: zivkan
ms.date: 2/4/2025
ms.topic: conceptual
---
# NuGet 6.13 Release Notes

<!--
    This is intentionally not yet added to TOC.md.
    When this version is ready for release:
    1. Rename this file to the next version
    2. Change the version strings in this file to the next version
    3. Use the release tool to create the real release notes for the version ready for release

>     4. Add the release notes to TOC.yml

Reviewer: The PR is not adding these release notes to the TOC.

<sup>1</sup> Installed with Visual Studio 2022 with any .NET workload

## Summary: What's New in 6.13

> * Detect if restore used NuGetAudit or not for PackageReference projects - [#13778](https://github.com/NuGet/Home/issues/13778)

Reviewer: We probably don't need this as a high level item and we can just move it to the issues fixed.

* Support for credential providers deployed via .NET tools - [#12567](https://github.com/NuGet/Home/issues/12567)

### Breaking changes

> * Support "Supplied by Platform" scenario in restore - [#7344](https://github.com/NuGet/Home/issues/7344)

Reviewer: This is not enabled in this release, so let's put this in `What's new`. 
I'll do a follow up for ensuring the pruning is enabled is documented.

> * Allow specifying the msbuild binlog path when invoking static graph restore to avoid modifying the environment - [#10789](https://github.com/NuGet/Home/issues/10789)

Reviewer: Did someone suggest this as a top level thing?
I don't see it as that impactful.

Author: https://github.com/NuGet/docs.microsoft.com-nuget/pull/3390#discussion_r1945140437

Reviewer: resolving.

> ### New Dependency Resolver Fixes

Author: Under Issues Fixed, an indented section for the new resolver fixes sounds good to me

> * New Dependency Resolver Fixes

Reviewer: I was thinking this can be in the `What's new section`.

* Navigation telemetry for 'Clear All NuGet Storage' Command in VS Settings - [#13968](https://github.com/NuGet/Home/issues/13968)

* New dependency resolver downloads more packages than before - [#13943](https://github.com/NuGet/Home/issues/13943)

* Nuget Packages Manager screen for Solution automatically selects transitive dependencies - [#13893](https://github.com/NuGet/Home/issues/13893)

* Debugging large repos with static graph-based restore is slow - [#13876](https://github.com/NuGet/Home/issues/13876)

* NuGet Client SDK packages updating to net8.0 - [#13842](https://github.com/NuGet/Home/issues/13842)

* Promote NU3043 warning to error in .NET 10 - [#13814](https://github.com/NuGet/Home/issues/13814)

> * dotnet nuget why should check RID specific packages - [#13718](https://github.com/NuGet/Home/issues/13718)

Reviewer: I want to nominate this for elevating it to the what's new section.

It's sparse right now, so this is important.

> * Allow specifying the msbuild binlog path when invoking static graph restore to avoid modifying the environment - [#10789](https://github.com/NuGet/Home/issues/10789)

Reviewer 2: This should be moved to What's New

* Setting "Allow format selection on first package install" meaning is unclear - [#14016](https://github.com/NuGet/Home/issues/14016)

> * New dependency resolver does not handle floating versions correctly in some situations - [#13992](https://github.com/NuGet/Home/issues/13992)

Reviewer: I wonder if we should create a collection of `New resolver fixes` in the what's new.

Each individual fix is probably not important enough, but the collection is. 

@jeffkl what do you think?

I can take a stab at creating a new line item for that.

Reviewer 2: Yeah I guess we might as well if its not too much work.

Reviewer: @Nigusu-Allehu can you do this or do you need help from us?

Author: On it

<sup>1</sup> Installed with Visual Studio 2022 with any .NET workload

> ## Summary: What's New in 6.13

Reviewer 3: I just relaised there are 6.13.0 and 6.13.1 releases going out.  I think we should separate them, tell customers which versions of VS & .NET SDK they ship in, and what the differences are between the two.

Reviewer 4: Yes, this is what we do usually.

* Support for credential providers deployed via .NET tools - [#12567](https://github.com/NuGet/Home/issues/12567)

> * Support "Supplied by Platform" scenario in restore - [#7344](https://github.com/NuGet/Home/issues/7344)

Reviewer 4: ```suggestion
* Opt-in feature: "Supplied by Platform", which removes packages that are supplied by the .NET platform from the dependency graph. This results in better performance and eliminates false positives for vulnerabilities in transitive dependencies. 
```

* Allow specifying the msbuild binlog path when invoking static graph restore to avoid modifying the environment - [#10789](https://github.com/NuGet/Home/issues/10789)

* New Dependency Resolver Fixes

  * New dependency resolver downloads more packages than before - [#13943](https://github.com/NuGet/Home/issues/13943)

  * New dependency resolver does not handle floating versions correctly in some situations - [#13992](https://github.com/NuGet/Home/issues/13992)

  * New dependency resolver erroneously reports NU1605 (downgrade) when using transitive pinning a direct dependency and a downgrade exists in a package graph - [#13938](https://github.com/NuGet/Home/issues/13938)

  * NuGet Restore restoring old versions of transitive dependencies when direct dependency does not have guidelines for user's targeted .NET Framework - [#13934](https://github.com/NuGet/Home/issues/13934)

  * Project and package in the same graph with the same name but different dependencies may lead to incorrect versions of the dependencies of that id  - [#13888](https://github.com/NuGet/Home/issues/13888)

> * Updated the Package Manager UI to render package README if they are available - [#12583](https://github.com/NuGet/Home/issues/12583)

Reviewer 4: ```suggestion
* Package Manager UI in Visual Studio now shows embedded READMEs for NuGet packages, if available - [#12583](https://github.com/NuGet/Home/issues/12583)
```

  * New dependency resolver erroneously reports NU1605 (downgrade) when using transitive pinning a direct dependency and a downgrade exists in a package graph - [#13938](https://github.com/NuGet/Home/issues/13938)

  * NuGet Restore restoring old versions of transitive dependencies when direct dependency does not have guidelines for user's targeted .NET Framework - [#13934](https://github.com/NuGet/Home/issues/13934)

  * Project and package in the same graph with the same name but different dependencies may lead to incorrect versions of the dependencies of that id  - [#13888](https://github.com/NuGet/Home/issues/13888)

### Issues fixed in this release

* Detect if restore used NuGetAudit or not for PackageReference projects - [#13778](https://github.com/NuGet/Home/issues/13778)

* Add indicator for deprecated and vulnerable packages to Package Details tab header. - [#13974](https://github.com/NuGet/Home/issues/13974)

* Navigation telemetry for 'Clear All NuGet Storage' Command in VS Settings - [#13968](https://github.com/NuGet/Home/issues/13968)

> * Nuget Packages Manager screen for Solution automatically selects transitive dependencies - [#13893](https://github.com/NuGet/Home/issues/13893)

Reviewer 5: ```suggestion
* Nuget Package Manager for Solution automatically selects transitive dependencies - [#13893](https://github.com/NuGet/Home/issues/13893)
```

* Debugging large repos with static graph-based restore is slow - [#13876](https://github.com/NuGet/Home/issues/13876)

* NuGet Client SDK packages updating to net8.0 - [#13842](https://github.com/NuGet/Home/issues/13842)

* Promote NU3043 warning to error in .NET 10 - [#13814](https://github.com/NuGet/Home/issues/13814)

* Setting "Allow format selection on first package install" meaning is unclear - [#14016](https://github.com/NuGet/Home/issues/14016)

> * [Bug Bash] The “README” tab doesn’t show in the right panel of the PM UI after installing package - [#13924](https://github.com/NuGet/Home/issues/13924)
> 
> * [Bug Bash] The tab “README” & “Package Details” should left-align with the content of each tab in the detail panel of PM UI  - [#13910](https://github.com/NuGet/Home/issues/13910)

Reviewer 5: Consider making the quotes "
```suggestion
* [Bug Bash] The "README" tab doesn’t show in the right panel of the PM UI after installing package - [#13924](https://github.com/NuGet/Home/issues/13924)

* [Bug Bash] The tab "README" & "Package Details" should left-align with the content of each tab in the detail panel of PM UI  - [#13910](https://github.com/NuGet/Home/issues/13910)
```

<sup>1</sup> Installed with Visual Studio 2022 with any .NET workload

## Summary: What's New in 6.13.1

> NuGet 6.13.1 is available in Visual Studio 17.13.

Reviewer 3: It's also available in .NET 9.0.201, even if it hasn't shipped yet

Author: I was thinking it would make sense if we update it when it ships.

> * Support for new slnx solution format - [#14034](https://github.com/NuGet/Home/issues/14034)

Reviewer 3: ```suggestion
* Support for new slnx solution format in dotnet nuget why and dotnet list package - [#14034](https://github.com/NuGet/Home/issues/14034)
```

I guess that dotnet and msbuild restore will also support slnx, but that's a function of msbuild, not NuGet.  However, static graph restore uses MSBuild APIs directly, and does not support slnx, so this release note shouldn't imply that Nuget supports slnx everywhere.

> ## Summary: What's New in 6.13

Reviewer 3: ```suggestion
## Summary: What's New in 6.13.0
```
